raft/raft.go

```diff
@@ -632,7 +632,7 @@ func stepLeader(r *raft, m pb.Message) {
 			r.id, r.raftLog.lastTerm(), r.raftLog.lastIndex(), r.Vote, m.From, m.LogTerm, m.Index, r.Term)
 		r.send(pb.Message{To: m.From, Type: pb.MsgVoteResp, Reject: true})
 	case pb.MsgSnapStatus:
-		if pr.State != ProgressStateSnapshot {
+		if pr == nil || pr.State != ProgressStateSnapshot {
```

Comment:
I'd love to just move the check at 569 up, but that's looking for r.id and not m.From

Also, only 5 out of the 8 messages types, look at pr.
- MsgProp
- MsgAppResp
- MsgHeartbeatResp
- MsgSnapStatus
- MsgUnreachable

While these don't:
- MsgVote
- MsgCheckQuorum
- MsgBeat

I'll add the check for each of the ones that rely on it.